### PR TITLE
trig output fixes

### PR DESCRIPTION
**Pull request by drewp**

### CODE REVIEW

File: rdflib/plugins/serializers/trig.py [hunk 1]
```diff
@@ -40,22 +41,6 @@ def preprocess(self):
...
-            if node in self.keywords:
-                continue
-            # Don't use generated prefixes for subjects and objects
-            self.getQName(node, gen_prefix=(i == VERB))
-            if isinstance(node, Literal) and node.datatype:
-                self.getQName(node.datatype, gen_prefix=_GEN_QNAME_FOR_DT)
-        p = triple[1]
-        if isinstance(p, BNode):
-            self._references[p]+=1
-
```
Comment (joernhees) [inline rdflib/plugins/serializers/trig.py]: did you remove this on purpose? seems it is called from `preprocess` and the one in `TurtleSerializer` looks quite different to me

Comment (drewp) [inline rdflib/plugins/serializers/trig.py]: TurtleSerializer's one is a super call followed by a repeat of lines 48-57. The super call would run RecursiveSerializer's method, which is a repeat of lines 45-47. Most importantly, the test suite still passes :)

Comment (joernhees) [inline rdflib/plugins/serializers/trig.py]: ah sorry, i must have looked at the `RecursiveSerializer` one directly... so essentially this was unnecessary code duplication?

File: rdflib/plugins/serializers/trig.py [hunk 2]
```diff
@@ -83,7 +68,10 @@ def serialize(self, stream, base=None, encoding=None,
             if self.default_context and store.identifier==self.default_context:
                 self.write(self.indent() + '\n{')
             else:
-                self.write(self.indent() + '\n<%s> {' % self.getQName(store.identifier))
+                iri = self.getQName(store.identifier)
+                if iri is None:
+                    iri = '<%s>' % store.identifier
+                self.write(self.indent() + '\n%s {' % iri)
```
Comment (joernhees) [inline rdflib/plugins/serializers/trig.py]: two questions here:
 - should it be possible that `store.identifier` isn't in `getQName` at all?
 - wouldn't this introduce a bug if `iri is not None` cause it's not wrapped in `<%s>` anymore?
    (so shouldn't line 73 be: `iri = '%s' % store.identifier` and 74: `self.write(self.indent() + '\n<%s> {' % iri)` ?

File: rdflib/plugins/serializers/trig.py (hunk 1 above)
Comment (drewp) [inline rdflib/plugins/serializers/trig.py]: Yes I think so

File: rdflib/plugins/serializers/trig.py (hunk 2 above)
Comment (drewp) [inline rdflib/plugins/serializers/trig.py]: getQName gets to return None if it has no abbreviation, which I test with the URIRef("http://example.com/foo.") case.

As I understand it, a non-None iri (e.g. a successful abbreviation like "ns1:foo") shouldn't be wrapped in <>. That seemed to be a bug before.  See Example 1 in http://www.w3.org/TR/trig/#sec-triple-statements for a demo of no angle brackets.

I pushed an update to get bnode graph ids right too (like "_:a1b2c3 { ... }").

**Comment by joernhees** [comment]
thanks for the changes and comments, i think this looks good now...

**Pull request merged**